src/com/palmergames/bukkit/towny/object/metadata/CustomDataField.java

```diff
@@ -163,13 +163,13 @@ public int hashCode() {
     public CustomDataField newCopy() {
         switch (type) {
             case BooleanField:
-                return new BooleanDataField(getKey(), (Boolean)getValue());
+                return new BooleanDataField(getKey(), (Boolean)getValue(), label);
             case IntegerField:
-                return new IntegerDataField(getKey(), (Integer)getValue());
+                return new IntegerDataField(getKey(), (Integer)getValue(), label);
             case DecimalField:
-                return new DecimalDataField(getKey(), (Double)getValue());
+                return new DecimalDataField(getKey(), (Double)getValue(), label);
             case StringField:
-                return new StringDataField(getKey(), (String)getValue());
+                return new StringDataField(getKey(), (String)getValue(), label);
```

Comment:
I was looking at this PR, and this is one of the major things that confused me. This PR allows metadata to use generics essentially allowing developers to create metadata with any value. However, it seems like this is still confining metadata objects into (essentially) primitive categories. 
Furthermore, these classes are simply wrapper classes and don't really provide any purpose (the parent class, customdatafield holds a type enum so that eliminates the only real use case for these empty wrapper classes which would be instance checking the class).

Reply:
Yeah, this is more of a past artifact, as a result the metadata system is prepared for custom objects from developers, it's mostly a placeholder for now until I implement it. Hopefully, the enum type checking will be temporary.

On that note,
I'm guessing what your saying is to have any objects which inherit from the `CustomDataField` to define their own way of saving and loading, as opposed to having a central enum to compare against. Thus allowing the class to actually be *dynamic*.